src/praisonai/tests/unit/test_approval_basic.py

```
@@ -0,0 +1,265 @@
+        print(f"⚠️ Shell tools test failed: {e}")
+        
+    # Test python tools
+    try:
+        from praisonaiagents.tools.python_tools import PythonTools
+        python_tools = PythonTools()
+        
+        # Check if execute_code requires approval
+        assert is_approval_required("execute_code"), "execute_code should require approval"
+        print("✅ PythonTools.execute_code requires approval")
+        
+    except Exception as e:
```

Pylint (3.3.7)

[warning] Catching too general exception Exception

(W0718)